src/functional/iterator/embedded.py

```diff
@@ -140,21 +141,29 @@ def can_deref(it):
 
 @builtins.if_.register(EMBEDDED)
 def if_(cond, t, f):
+    if isinstance(cond, np.ndarray):
```

Comment:
I went with the np.ndarray approach, but could make sense to introduce a column type as you suggested, Till.